docs/migration-guides/v4.md

```diff
@@ -0,0 +1,327 @@
+# Migrating to Tool Kit v4
+
+Version 4 of Tool Kit lets you configure it much more easily for your team's or project's specific use cases, without having to resort to custom plugins as often. Although we've made it as backwards compatible as possible, there are some fundamental changes to its internals and configuration that mean you'll need to update your repo's `.toolkitrc.yml` and any custom plugins when migrating to Tool Kit v4.
+
+> [!NOTE]
+> This migration will take 30m-1hr. If at any point you get stuck or something unexpected happens, [reach out to the Platforms team](https://financialtimes.enterprise.slack.com/archives/C02TRE2V2Q1) and we'll be happy to help.
+
+## 0. Update to latest versions of Tool Kit packages
+
+Update `dotcom-tool-kit` and all the `@dotcom-tool-kit/*` packages you have installed to their latest versions. This can be done with the following command (requires [`jq`](https://jqlang.github.io/jq/)):
+
+```sh
+npm query ":root > [name*=dotcom-tool-kit]" \
+| jq '(.[] | .name + "@latest")' \
+| xargs npm install
+```
+
+<details><summary>what</summary>
+
+1. `npm query ":root > [name*=dotcom-tool-kit]"`: get all packages installed as a direct dependency of your repo with `dotcom-tool-kit` in their name
+2. `jq '(.[] | .name + "@latest")'`: output a list of just the `name` from the packages, appending `@latest`
+3. `xargs npm install`: pass those as arguments to the `npm install` command
+
+So if you have `dotcom-tool-kit`, `@dotcom-tool-kit/jest` and `@dotcom-tool-kit/webpack` installed, this is equivalent to running:
+
+```
+npm install dotcom-tool-kit@latest @dotcom-tool-kit/jest@latest @dotcom-tool-kit/webpack@latest
+```
+
+(but without you having to manually type that out).
+
+</details>
+
+## 1. Make required changes to configuration
+
+### 1.1. Rename `hooks` to `commands`
+
+> [!NOTE]
+> **Why?** In previous versions of Tool Kit, hooks both defined names of commands to run, and managed configuration files. In Tool Kit v4, hooks only manage configuration files, and commands are defined in the Tool Kit configuration files, not in plugin code.
```

Comment:
**suggestion**: think here would be a good place to expand on what we mean by a command

```suggestion
> **Why?** In previous versions of Tool Kit, hooks both defined names of commands to run, and managed configuration files. In Tool Kit v4, hooks only manage configuration files, and commands are defined in the Tool Kit configuration files, not in plugin code. Commands comprise of an explicitly ordered set of tasks to run.
```